### ISSUE TITLE
Question about fine-tuning a pretrained model

### ISSUE THREAD

**Issue by Maxwell-downtown** [gitter-lab/metl#13]
Very nice work. 
I am trying to pretrain the model from scratch and then fine-tune it on experimental data, following the workflow described in the repository.

Specifically, I used the AvgFP example dataset provided in the repository and pretrained the source model using the command shown in the README:
```
python code/train_source_model.py @args/pretrain_avgfp_local.txt
```
After training completed, I obtained a checkpoint at: `output/training_logs/nuUzH9oA/checkpoints/last.ckpt`. I then modified `args/finetune_avgfp_local.txt` by seting the `--pretrained_ckpt_path` argument to point to this checkpoint, and proceeded to fine-tune the model on the experimental data using:
```
python code/train_target_model.py @args/finetune_avgfp_local.txt
```
During the fine-tuning step, I encountered the following warning and error related to distributed training:
```
Warning: find_unused_parameters=True was specified in DDP constructor, but did not find any unused parameters in the forward pass. This flag results in an extra traversal of the autograd graph every iteration, which can adversely affect performance. If your model indeed never has any unused parameters in the forward pass, consider turning this flag off. Note that this warning may be a false positive if your model has flow control causing later iterations to have unused parameters. (function operator())

pytorch_lightning.utilities.exceptions.MisconfigurationException: 
`return_predictions` should be set to `False` when using the `DDPSpawnStrategy` or children class. 
Found True with strategy <class 'pytorch_lightning.strategies.ddp_spawn.DDPSpawnStrategy'>.
```
Despite this error message, the script does continue running and outputs test metrics. However, I noticed that the fine-tuned model’s performance is substantially lower than expected. In my runs, the test Spearman correlation is typically around 0.25–0.30, whereas the performance reported on the GitHub page for AvgFP appears to be around 0.6. This makes me wonder whether I may be missing some important configuration when fine-tuning from a self-pretrained checkpoint. In particular, I am not sure whether additional arguments in args/finetune_avgfp_local.txt need to be modified beyond --pretrained_ckpt_path, or whether there are other recommended changes (either in the configuration files or in the code) that are necessary to correctly fine-tune a pretrained source model.

Meanwhile, these warning and errors won't show up if i use the original `finetune_avgfp_local.txt` file with the preset model checkpoint directory under `pretrained_models/Hr4GNHws.pt`

Any guidance on these points would be greatly appreciated. Thank you very much for your time and for developing such a useful framework.

**Comment by agitter** [comment]
We can help look into this. Could you first confirm what software environment you're using? Did you create a conda environment from `environment.yml`?

**Comment by Maxwell-downtown** [comment]
Thanks for your help.
I think I create the virtual environment with mamba, using the following command:
`mamba env create -f environment.yml`

**Comment by Maxwell-downtown** [comment]
(By the way, I checked the package information of the environment using `mamba list`, and all package versions appear to be consistent with those specified in the provided `environment.yml` file. I did not notice any version mismatches or unexpected dependencies introduced during environment creation.)

**Comment by agitter** [comment]
I ran a new [test job](https://github.com/gitter-lab/metl/actions/runs/21648571046/job/62406896169#step:6:122), which runs this same pretrain and finetune process on avGFP. The test Spearman correlation was 0.55, so closer to our previously reported value and different from your results.

Could you also please tell us about your hardware setup? Is this on CPU, GPU, or other? If GPU, is it multi-GPU?

I'll take that information to my lab members who are actively using METL to give you advice and patch our code if needed. You could also try pretraining and finetuning with CPU-only if you haven't done that already.

**Comment by Maxwell-downtown** [comment]
Thanks for your reply. We reran pretraining + fine-tuning with CPU (and also single-GPU). In both cases, the DDPSpawnStrategy-related error disappears, but the test Spearman correlation stayed around 0.25–0.30.

We then tuned a few settings in args/pretrain_avgfp_local.txt (lower learning rate, higher max_epochs, and LR scheduler tweaks). With these changes, the fine-tuned test Spearman improved to 0.5–0.6, close to the reported result. I guess the default `args/pretrain_avgfp_local.txt` might be an earlier configuration and might be the cause for the lower spearman correlation we initially observed.

**Comment by agitter** [comment]
Thanks for following up @Maxwell-downtown. That's good news you were able to match performance, but still leaves some unresolved questions for us to ponder on our side.

Our integration tests run
```
python code/train_source_model.py @args/pretrain_avgfp_local.txt --max_epochs 5 --limit_train_batches 5 --limit_val_batches 5 --limit_test_batches 5
```
using `args/pretrain_avgfp_local.txt`, so I wouldn't expect that file to be stale.
